server/fleet/windows_mdm.go

```diff
@@ -56,44 +65,69 @@ func (m *MDMWindowsConfigProfile) ValidateUserProvided() error {
 		return fmt.Errorf("Profile name %q is not allowed.", m.Name)
 	}
 
-	var validator struct {
-		SyncBody
-		NonProtocolElements []interface{} `xml:",any,omitempty"`
-	}
-	wrappedProfile := fmt.Sprintf("<SyncBody>%s</SyncBody>", m.SyncML)
-	if err := xml.Unmarshal([]byte(wrappedProfile), &validator); err != nil {
-		return fmt.Errorf("The file should include valid XML: %w", err)
-	}
-
-	// might be valid XML, but start with something other than <Replace>
-	if mdm.GetRawProfilePlatform(m.SyncML) != "windows" {
-		return errors.New("Only <Replace> supported as a top level element. Make sure you don't have other top level elements.")
-	}
+	dec := xml.NewDecoder(bytes.NewReader(m.SyncML))
+	// use strict mode to check for a variety of common mistakes like
+	// unclosed tags, etc.
+	dec.Strict = true
+
+	// keep track of certain elements to perform Fleet-validations.
+	//
+	// NOTE: since we're only checking for well-formedness
+	// we don't need to validate the required nesting
+	// structure (Target>Item>LocURI) so we don't need to track all the tags.
+	var inReplace bool
+	var inLocURI bool
+
+	for {
+		tok, err := dec.Token()
+		if err != nil {
+			if err != nil {
+				if err != io.EOF {
+					return fmt.Errorf("The file should include valid XML: %w", err)
+				}
+				break
```

Comment:
It took me a minute to follow the loop all the way through back to this break so I though it might be helpful to add a comment.

```suggestion
                                 // EOF means no more tokens to process
				break
```